src/source.rs

```diff
@@ -320,6 +327,11 @@ mod tests {
         async fn read(&self, request: SourceReadRequest, transmitter: Sender<Message>) {
             let event_time = Utc::now();
             let mut message_offsets = Vec::with_capacity(request.count);
+            let mut headers = HashMap::new();
+            let header_key=String::from(Uuid::new_v4());
+            let header_value = String::from("numaflow");
+            headers.insert(header_key, header_value);
```

Comment:
```suggestion
            let mut headers = HashMap::new();
            let header_key=String::from("x-txn-id");
            let header_value = String::from(Uuid::new_v4());
            headers.insert(header_key, header_value);
```

also move the headers inside the for loop, ideally each message will have different txn-id